### ISSUE TITLE
Running with -evaluate repeatedly performs the same action

### ISSUE THREAD

**Issue by tpbarron** [watts4speed/fast-dqn-caffe#5]
@watts4speed Have you had any issues running in evaluate mode? When testing with a well-trained network I'm getting the same set of q-values repeatedly for entire episodes with very little change and, as a result, the same actions. Have you seen this problem? If this is a training problem do you have any suggestions for improvement? I've been running tests for 24 hours and the error plot looks decent. 
Thanks, Trevor

**Comment by watts4speed** [comment]
Hi Trevor,

Are you running the scripts/plot.py when your training?  You should see the
scores improve.  When you then run in evaluation mode you should see the
same approximate score.

I haven't implemented an evaluation during the training which should really
be there.  The current implementation is a bit limited in that it uses the
score of a training run to show the progress of the training.  A better way
to do this is to play a separate evaluation episode every N training
episodes to see how things are training up.  The current implementation
plots a smoothed set of training scores to try and pull a signal out of a
very noisy set of scores during training which is also useful.

**Comment by tpbarron** [comment]
Yeah. I'm running scripts/plot.py during training and the score clearly increases. But when loading one of the snapshotted models I don't get scores anywhere close to what the plot reports during training.

**Comment by watts4speed** [comment]
That's strange, other than maybe something to do with the way the
evaluation is being run.  I haven't run the evaluation much at all
recently.  There is some difference in the way the actions are being chosen
during evaluation.  I think there's a parameter to choose the elision for
doing the evaluation.  I'd have to go look at the code again, but during
evaluation I think you want only use the "net" to choose commands.  If
epsilon-gready is being used then there's a randomness being introduced
which could lower the score.

**Comment by tpbarron** [comment]
The problem was really simple. It turned out that when loading a model the weights were copied to the training net but not the target net so the actions were really being chosen from a randomly initialized network. 

Just adding a line, `CloneTrainingNetToTargetNet();` to the `LoadTrainedModel()` method worked fine.

**Comment by watts4speed** [comment]
Cool!  Thanks for finding this!
